src/commands/roll.ts

```diff
@@ -1,94 +1,133 @@
 import { Client, Message } from 'discord.js';
 
 exports.run = async (client: Client, msg: Message, args: string[]) => {
-  // Joins args together in case spaces are in the expression
-  const diceEq = args.join('').replace(' ', '').toLowerCase();
+  try {
+    // Joins args together in case spaces are in the expression
+    const diceEq = coerceDiceEq(args.join(''));
 
-  const result = roll(diceEq);
+    await validate(diceEq);
 
-  msg.channel.send(result);
+    const statements = await parse(diceEq);
+
+    const result = await roll(statements);
+
+    msg.channel.send(result);
+  } catch (e) {
+    msg.channel.send((e as Error).message);
+  }
 };
 
 exports.help = {
-  description: 'Roll some dice! Input any dice euqation and recieve a result.',
+  // tslint:disable-next-line: max-line-length
+  description: 'Roll some dice! Input a dice euqation and recieve a result. Equations may only contain addition & subtraction and numbers can\'t be larger then 3 digits long',

```

Comment:
Minor spell checking for equation and receive here.

Reply:
me write code, no know spell word gud